app/lib/task/backend.dart

```diff
@@ -0,0 +1,881 @@
+      }
+      final versionState = state.versions![version]!;
+
+      // Check the secret token
+      if (!versionState.isAuthorized(_extractBearerToken(request))) {
+        throw AuthenticationException.authenticationRequired();
+      }
+      assert(versionState.scheduled != DateTime(0));
+      assert(versionState.instance != null);
+      assert(versionState.zone != null);
+
+      zone = versionState.zone!;
+      instance = versionState.instance!;
+
+      // Remove instanceName, zone, secretToken, and set attempts = 0
+      state.versions![version] = PackageVersionState(
+        scheduled: versionState.scheduled,
+        attempts: 0,
+        instance: null, // version is no-longer running on this instance
+        secretToken: null, // TODO: Consider retaining this for idempotency
+        zone: null,
+      );
+
+      // Determine if something else was running on the instance
+      isInstanceDone = state.versions!.values.none(
+        (v) => v.instance == instance,
+      );
+
+      // Update dependencies, if pana summary has dependencies
+      final summary = await panaSummary(package, version);
+      if (summary != null && summary.allDependencies != null) {
+        // TODO: Limit size of dependencies!!
+
+        final dependencies = {...state.dependencies ?? []};
+        // Only update if new dependencies have been discovered.
+        // This avoids unnecessary churn on datastore when there is no changes.
+        if (!dependencies.containsAll(summary.allDependencies ?? [])) {
+          state.dependencies = {
+            ...dependencies,
+            ...summary.allDependencies ?? [],
+          }.sorted();
+        }
+      }
+
+      tx.insert(state);
+    });
+
+    // If nothing else is running on the instance, delete it!
+    // We do this in a microtask after returning, so that it doesn't slow down
+    // worker response. We avoid doing it in the transaction because we wish to
+    // avoid doing this operation again if the transaction fails.
+    if (isInstanceDone) {
+      assert(zone != null && instance != null);
+      _log.fine('instance $instance is done, calling APIs to terminate it!');
+      scheduleMicrotask(() async {
+        try {
+          await _cloudCompute.delete(zone!, instance!);
+        } catch (e, st) {
+          _log.severe(
+            'failed to delete task-worker w. zone/instance: $zone/$instance',
+            e,
+            st,
+          );
+        }
+      });
+    }
+
+    // Clear cache entries for package / version
+    await _purgeCache(package, version);
+
+    return shelf.Response.ok('');
+  }
+
+  Future<List<int>?> _readFromBucket(
+    String path, {
+    int? offset,
+    int? length,
+  }) async =>
+      await retry(
+        () async {
+          try {
+            return await readChunkedStream(
+              _bucket.read(path, offset: offset, length: length),
+              maxSize: 10 * 1024 * 1024, // sanity limit
```

Comment:
No, that's an extra round-trip, it's possible we should increase the limit.

I just wanted to have some sanity limit, since this is a file generated by the untrusted sandbox.